src/Controls/src/Core/Handlers/Items2/iOS/ItemsViewController2.cs

```diff
@@ -663,3 +702,19 @@ internal virtual void CellDisplayingEndedFromDelegate(UICollectionViewCell cell,
 		}
 	}
 }
+
+internal static class CollectionViewMeasurementCache
+{
+	internal static readonly BindableProperty FirstItemMeasuredSizeProperty =
```

Comment:
This attached properties store data on the BindableObject (CollectionView) and the data persists for the lifetime of the CollectionView. How impact in the memory usage? CollectionView memory usage is really important. 
Can store in the handler, and correctly clear cache when disconnecting etc.?